platform/engineering/design-docs/2022-09-06-mobile-filtering.md

```diff
@@ -0,0 +1,162 @@
+  - No versioning of endpoints required.
+  - We know it works and doesn't yield any big surprises.
+  - Vets-api currently implementation of this uses query params so some logic could be borrowed from this.
+  - Sticks with convention of other mobile endpoints.
+
+#### Cons
+  - Longer, hard to read urls for requests with multiple filter parameters. URL length limits depend on browswers and webservers. Browswers are not relevant for our purposes, and popular webservers like nginx and apache support at least 4000 characters. Encoding special characters used in the filter query params will use some of that space, but it's unlikely that the 4000 character limit will be a problem.
+  - Cannot easily support complex filtering requirements. Some operations could be created for more complex data types, such as arrays and hashes, but even simple equality operators could become problematically complex for non-primitive data types.
+
+#### Overview
+The support of complex filtering requirements listed in the cons is not a requirement of this feature, nor does it seem practical for it to become a requirement for these endpoints in the future. Harder to read URLs is not a significant concern. Long and complex urls are widely used in the modern web and will not be visible to mobile users. Unless we anticipate very complex queries, length should not be a concern.
+
+On a side note, there is also an uncommonly used HTTP QUERY method that is exactly what we're looking for. It's essentially a GET with a body. However, as far as I can tell, it is not supported by rails.
+
+### GET with Payload Body
+#### Pros
+  - Allows for more robust filtering logic
+  - Keeps the url parameters from getting too long and hard to read
+  - Backwards compatible.
+  - No versioning of endpoints required.
+
+#### Cons
+  - Breaks REST/HTTP specs. While GET technically allows to have a payload body, it is not supposed to have any semantic effect on the request itself, that would not be true in this case.
+
+#### Overview
+Breaking REST/HTTP spec is too dangerous and may have too many unintended consquences to implement, and the additional flexibility it offers is not a current or expected requirement.
+
+### POST with Payload
+#### Pros
+  - Allows for more robust filtering logic
+  - Keeps the url parameters from getting too long and hard to read
+
+#### Cons
+  - Would require new endpoint for each Index action. This should not be a significant problem because the new endpoint can point to the same method as the existing GET endpoint.
+  - Take the most time to implement and integerate into FE.
+
+#### Overview
+Given that support of complex filtering requirements listed in the cons is not a requirement of this feature, nor does it seem practical for it become a requirement for these endpoints in the future, it does not seem worth it to take the extra time to implement this.
+
+### Conclusion
+We recommend using query params to transmit filter options.
```

Comment:
I talked to the front end team about this and Narin said he'd prefer the query params approach, so that's still the path forward.